src/osutil.c

```diff
@@ -63,10 +63,10 @@ void _sem_ev_post(SEMAPHORE_T *sem){
    SEMAPHORE_POST(*sem);
 #else
    int val=0;
-   int rc = sem_getvalue(sem,&val);
+   const int rc = sem_getvalue(sem, &val);
    if( (rc==0) && (val==0) ){
       sem_post(sem);
-      rc = sem_getvalue(sem,&val);
+      (void) sem_getvalue(sem, &val);
```

Comment:
This function is so many years old I almost forgot I wrote it.
Why is sem_getvalue called a second time after post? What possible side effect is that supposed to achieve? The getvalue function will write into the val variable which is on the stack so it gets thrown away.
Why not call sem_post and end it?

Reply:
Val should be allowed to be <=0 for full posix compliance

http://man7.org/linux/man-pages/man3/sem_getvalue.3.html

The second call must have only been for debugability? Ill update PR with these 2 changes...